chrome/content/addFinger.js

```
@@ -0,0 +1,34 @@
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("resource:///modules/imServices.jsm");
+Cu.import("resource:///modules/imXPCOMUtils.jsm");
+Cu.import("chrome://otr/content/otr.js");
+
+XPCOMUtils.defineLazyGetter(this, "_", () =>
+  l10nHelper("chrome://otr/locale/addfinger.properties")
+);
+
+var args = window.arguments[0].wrappedJSObject;
+
+var otrAddFinger = {
+  onload: function() {
+    document.title = _("addfinger.title", args.buddy);
+  },
+  
+  oninput: function(e) {
+    document.documentElement.getButton("accept").disabled = (e.value.length != 40);
+  },
+
+  add: function(e) {
+    let hash = document.getElementById("finger").value;
+    let context = otr.getContextFromRecipient(
+          args.account,
+          args.protocol,
+          args.screenname
+        );
+    let fingerprint = otr.addFingerprint(context, hash);
```

Oop, I see the `oninput` above now. Well then `addFingerprint` should throw instead of returning `null`.